### PR TITLE
proxy support for mvn project

### PR DESCRIPTION
**Pull request by idlelearner**
Ref : https://algorithmia.atlassian.net/browse/DEPLOY-2

### CODE REVIEW

**Comment by idlelearner** [comment]
Ok, let me check about the changes needed for java runtime and scala runtime. As far as I researched, for `mvn package`, we need to set the proxy host either in the .m2/settings.xml or pass as java runtime ops in cmdline invocation.

**Comment by idlelearner** [comment]
sbt would use JAVA_OPTS while building but for mvn we need to set it as mentioned above

**Comment by kennydaniel** [comment]
Seems reasonable to me to make it respect the JAVA_OPTS flag

File: java/template/bin/build [hunk 1]
```diff
@@ -16,6 +20,36 @@ if [[ -n "$MAVEN_MIRROR" ]]; then
     </mirror>
   </mirrors>
 
+EOF
+fi
+
+if [[ -n "$HTTPS_PROXY" ]]; then
+        cat >> ~/.m2/settings.xml << EOF
```
[the remark this answers is not in the document]
Comment (pmcq) [reply java/template/bin/build]: :shrug: i think it was for supporting private repos in specific enterprise deploys. @jamesatha

**Pull request merged**